src/app/profile/yourGiving/stopStartRecurringGifts/restartGift/step3/confirmGifts/confirmGifts.component.js

```diff
@@ -0,0 +1,75 @@
+import angular from 'angular';
+import template from './confirmGifts.tpl';
+import isEmpty from 'lodash/isEmpty';
+import concat from 'lodash/concat';
+import {Observable} from 'rxjs/Observable';
+import 'rxjs/add/observable/forkJoin';
+import 'rxjs/add/operator/do';
+
+import giftListItem from 'common/components/giftViews/giftListItem/giftListItem.component';
+import giftDetailsView from 'common/components/giftViews/giftDetailsView/giftDetailsView.component';
+
+import donationsService from 'common/services/api/donations.service';
+
+let componentName = 'confirmGifts';
+
+class ConfirmGiftsController {
+
+  /* @ngInject */
+  constructor( donationsService ) {
+    this.donationsService = donationsService;
+  }
+
+  $onInit() {
+    this.updates = [];
+    this.adds = [];
+    this.saved = [];
+    angular.forEach( this.gifts, ( gift ) => {
+      (angular.isDefined( gift.parentDonation['donation-status'] ) ? this.updates : this.adds).push( gift );
+    } );
+  }
+
+  processRestarts() {
+    let requests = [];
+    this.setLoading( {loading: true} );
+    if ( !isEmpty( this.adds ) ) {
+      requests.push( this.donationsService.addRecurringGifts( this.adds ).do( () => {
+        this.saved = concat( this.saved, this.adds );
+        this.adds.length = 0;
```

Comment:
Can you make sure this works if the user saves, has one set of gift succeed and the other fail, and then goes back to some other step and then comes back to confirm?  Hopefully it is an edge case but I think it would be bad if a user ended up adding gifts twice. Maybe you've already tested it or maybe it already works. In my modals, in one of the cases, going back would forget the changes done here (because it was using data in the parent component to send to this confirm page) and have the gifts added back when they came back.

An easy way I found to test it is just to comment out the `this.next()` below.